edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/handlers/ConnectionHandler.cs

```diff
@@ -122,11 +122,41 @@ public async Task CloseConnectionAsync(IIdentity identity)
             }
         }
 
-        public async Task<IReadOnlyList<IIdentity>> GetNestedConnectionsAsync()
+        public async Task UpdateNestedParentInfoAsync(IEnumerable<IIdentity> childDevices, IIdentity parentIdentity)
         {
             using (await this.guard.LockAsync())
             {
-                return this.knownConnections.Values.OfType<IDeviceProxy>().Where(p => !p.IsDirectClient).Select(p => p.Identity).ToArray();
+                foreach (var children in childDevices)
+                {
+                    var maybeListener = default(Option<IDeviceListener>);
+                    if (!this.knownConnections.TryGetValue(children, out var listener))
+                    {
+                        maybeListener = await this.CreateDeviceListenerAsync(children, false);
+                    }
+                    else
+                    {
+                        maybeListener = Option.Some(listener);
+                    }
+
+                    maybeListener.ForEach(
+                        l =>
+                        {
+                            if (l is IDeviceProxy proxy)
+                            {
+                                proxy.ConnectionInfo.BindToParent(parentIdentity);
```

Comment:
since this method is called inside the guard locked does is still need to use Interlocked.Exchange internally?

Reply:
Yes, but the ConnectionInfo technically an independent class, and although we know that this is the only use and a simple assignment and check at the previous value would work, still felt bad doing that. But maybe it is safe to remove (that is a check which should not flag in theory)

Reply:
ok, removed the interlocked and changed to 'pure' option